worker.rb

```diff
@@ -27,39 +29,51 @@
       key = URI.decode(record['s3']['object']['key']).gsub('+', ' ')
       log.info "scanning s3://#{bucket}/#{key}..."
       begin
-        resp = s3.get_object(
-          response_target: '/tmp/target',
+        res = s3.head_object(
           bucket: bucket,
           key: key
         )
+        filesize = res.content_length
+        status = 'OK'
+        log.info "s3://#{bucket}/#{key} file size : #{filesize}"
+        if filesize <= LIMIT_SIZE
+          log.info "s3://#{bucket}/#{key} is under the limitation"
+          resp = s3.get_object(
+            response_target: '/tmp/target',
+            bucket: bucket,
+            key: key
+          )
+          if system('clamscan /tmp/target')
+            log.info "s3://#{bucket}/#{key} was scanned without findings"
+            status = 'OK'
```

Comment:
Status is already set to ok line 37. I'm not sure that this line is condition is required.

Reply:
I moved the default value to be more accurate